src/expression/expression-builder.ts

```diff
@@ -150,6 +155,64 @@ export interface ExpressionBuilder<DB, TB extends keyof DB> {
     from: TE
   ): SelectQueryBuilder<From<DB, TE>, FromTables<DB, TB, TE>, {}>
 
+  /**
+   * Creates a `case` statement/operator.
+   *
+   * ### Examples
+   *
+   * Kitchen sink example with 2 flavors of `case` operator:
+   *
+   * ```ts
+   * import { sql } from 'kysely'
+   *
+   * const { title, name } = await db
+   *   .selectFrom('person')
+   *   .where('id', '=', '123')
+   *   .select((eb) => [
+   *     'first_name',
+   *     eb
+   *       .case()
+   *       .when(eb.cmpr('gender', '=', 'male'))
+   *       .then(sql.lit('Mr.'))
```

Comment:
The examples are full of `sql.lit` which feels wrong for some reason. I still don't feel good about adding `eb.lit` though 🤔 

I'm guessing `eb.val` doesn't work here without SQL casts?

Reply:
I tried to mimmick how I'd write this case example in SQL without a query builder. 
I'd use literals in WHEN and THEN unless its user input, because paramters here feel odd.

FYI, I actually wrote `'male'` because right now, `cmpr` doesn't accept `sql.lit` (which is an issue for another PR).